tests/unit/output/test_yaml_output.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import ruamel.yaml as yaml
+
+from awscli.testutils import BaseAWSCommandParamsTest
+from awscli.testutils import skip_if_windows
+
+
+class TestYAMLOutput(BaseAWSCommandParamsTest):
+
+    def setUp(self):
+        super(TestYAMLOutput, self).setUp()
+        self.parsed_response = {
+            'Users': [
+                {
+                    "UserName": "testuser-50",
+                    "Path": "/",
+                    "CreateDate": "2013-02-12T19:08:52Z",
+                    "UserId": "EXAMPLEUSERID",
+                    "Arn": "arn:aws:iam::12345:user/testuser1"
+                },
+                {
+                    "UserName": "testuser-51",
+                    "Path": "/",
+                    "CreateDate": "2012-10-14T23:53:39Z",
+                    "UserId": u"EXAMPLEUSERID",
+                    "Arn": "arn:aws:iam::123456:user/testuser2"
+                },
+            ]
+        }
+
+    def test_yaml_response(self):
+        output = self.run_cmd(
+            'iam list-users --output yaml', expected_rc=0
+        )
+        parsed_output = yaml.safe_load(output[0])
+        self.assertEqual(self.parsed_response, parsed_output)
+
+    def test_empty_dict_response_prints_nothing(self):
+        self.parsed_response = {}
+        stdout = self.run_cmd('sts get-caller-identity', expected_rc=0)[0]
+        self.assertEqual(stdout, '')
```

Comment:
Could you elaborate on why an empty dictionary is nothing but an empty list is still an empty list?

Reply:
Keeping the same behavior as the json formatter. I imagine we don't want to print emtpy dicts because then all commands would have output, even if they return nothing. To get a list you'd have to be querying. I would be +0 on not printing empty lists.